validator/sawtooth_validator/config/validator.py

```diff
@@ -0,0 +1,187 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import collections
+import logging
+import os
+
+import toml
+
+from sawtooth_validator.exceptions import LocalConfigurationError
+from sawtooth_validator.config.path import get_default_path_config
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def load_toml_validator_config(filename):
+    """Returns a ValidatorConfig created by loading a TOML file from the
+    filesystem.
+    """
+    if not os.path.exists(filename):
+        LOGGER.info(
+            "Skipping validator config loading from non-existent config file:"
+            " %s", filename)
+        return ValidatorConfig()
+
+    LOGGER.info("Loading validator information from config: %s", filename)
+
+    try:
+        with open(filename) as fd:
+            raw_config = fd.read()
+    except IOError as e:
+        raise LocalConfigurationError(
+            "Unable to load validator configuration file: {}".format(str(e)))
+
+    toml_config = toml.loads(raw_config)
+
+    invalid_keys = set(toml_config.keys()).difference(
+        ['network_endpoint', 'component_endpoint', 'public_uri', 'peering',
+         'join', 'peers'])
+    if len(invalid_keys) > 0:
+        raise LocalConfigurationError(
+            "Invalid keys in validator config: "
+            "{}".format(", ".join(sorted(list(invalid_keys)))))
+
+    peers = None
+    join = None
+    if "peers" in toml_config:
+        peers = toml_config["peers"].split(",")
+
+    if "join" in toml_config:
+        join = toml_config["join"].split(",")
+
+    config = ValidatorConfig(
+         network_endpoint=toml_config.get("network_endpoint", None),
+         component_endpoint=toml_config.get("component_endpoint", None),
+         public_uri=toml_config.get("public_uri", None),
+         peering=toml_config.get("peering", None),
+         join=join,
+         peers=peers,
+    )
+
+    return config
+
+
+def merge_path_config(configs):
+    """
+    Given a list of ValidatorConfig objects or command line optoins, merges
+    them into a single ValidatorConfig, giving priority in the order of the
+    configs (first has highest priority).
+    """
+    network_endpoint = None
+    component_endpoint = None
+    public_uri = None
+    peering = None
+    join = None
+    peers = None
+
+    for config in reversed(configs):
+        if config.network_endpoint is not None:
+            network_endpoint = config.network_endpoint
+        if config.component_endpoint is not None:
+            component_endpoint = config.component_endpoint
+        if config.public_uri is not None:
+            public_uri = config.public_uri
+        if config.peering is not None:
+            peering = config.peering
+        if config.join is not None:
+            join = config.join
+        if config.peers is not None:
+            peers = config.peers
```

Comment:
This might be cleaner in a loop. Might be good to make the list of options part of ValidatorConfig too. Something like:
```
merged = ValidatorConfig()
for config in reversed(configs):
  for field in config.list_fields():
    if config.getattr(field):
      merged.setattr(field, config.getattr(field))
```